### ISSUE TITLE
Acceptance testing: Normalize space on deposit.

### ISSUE THREAD

**Issue by Alexandermay** [curationexperts/epigaea#362]
End users are going to be copy and pasting titles, descriptions, etc into the fields, and this will result in strange spacing if it is not normalized.  As a result, catalogers will need to spend extra time cleaning it up.

Normalizing space on deposit should occur automatically.

See title field in: w9505044z

![screen shot 2017-10-04 at 8 27 29 am](https://user-images.githubusercontent.com/4700092/31175687-1b03e7e8-a8de-11e7-9eed-535ed3602d20.png)

**Comment by mark-dce** [comment]
@Alexandermay - does "normalize white space" mean "replace all occurrences of multiple spaces and/or tabs with a single space?"  If so, I think we can make that happen pretty easily; otherwise, we'll need some help on the normalization rules.

**Comment by Alexandermay** [comment]
Replace all occurrences of multiple spaces and/or tabs with a single space is exactly what I meant.

**Comment by mark-dce** [comment]
Great - we'll get that in shortly.

**Comment by mark-dce** [comment]
**ACCEPTANCE**
- [ ] Normalize whitespace on all self deposit fields

Normalize = collapse all consecutive whitespace (space, tab, carriage returns) into a single space character

**Comment by sarimauro** [comment]
As of Monday this was happening inconsistently (see 2r36tx62f). By our assessment, only multiple spaces in the middle of lines are being collapsed, not tabs or carriage returns.

**Comment by Alexandermay** [comment]
I just checked a title field and got the following result, and I'm pretty sure this duplicates the original request:
![Screen Shot 2018-03-14 at 3.23.28 PM.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59cd282390570f0012dbecea/125516c66e82c728ace21e0d46db978826878de9bce1b057e40fa0c5710b77206455f76b7191a37bf110284f5d575ecc3c4a4356bcbbd671f3e57862804e0ee4d0385f25c23b12f9b8864eea01b821834cd7b3b98818d2f42f67b6d5db1063474f4dcbed5727a19e41c95df44ae14850a110366be2732431.png)

**Comment by mark-dce** [comment]
@Alexandermay - that doesn't look like a self-deposit item (it's set to display in Trove and isn't marked as self-deposit). I had understood that Tufts did not want to normalize administrative deposits as the DCA and Tisch teams might want the flexibility to add manual formatting.

When I submit a work with that title through self deposit, the title is normalized as expected.  See the `title` and `description` here http://epigaea.curationexperts.com/concern/pdfs/j098zb217/edit

The above work looked like this in the submission form:
![EPIGAEA-362 Spacing follow-up submission.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/55118693805715190031f073/125516c66e82c728ace21e0d46cda4b304a3a688c2bae911e46ee0952a4f29746902ac277c8af465aa523b121a0d13f5021b061ba2eb9539e6f4632adf0f4fffdb7f5a7dc76811fdbc841cea01be71de4f87b7b4891ec5ed3f71f5979d0d6f405147d4a61760e28b41da59f141e04857a513336be775337443900c38.png)

**Comment by mark-dce** [comment]
After the initial round (where we collapsed all whitespace), Tufts determined that initial line indentations should be retained for formatting purposes and blank lines between paragraphs should be retained for specified fields.

So, in self deposit, the above screenshot yields
`title` 
```
Portrait of Fernando de Alencastre, Duke of Linares, Viceroy of Mexico.
```  

`description`
```
This line should not have spaces

This line should be preceded by a blank line.
 This line should be indented
 As should this one
```

The fields that we understood Tufts wished to have newlines preserved in are: `description` and `abstract`
This list can be updated in the codebase here: https://github.com/curationexperts/epigaea/blob/accdcd100f1cf52fdd221c061a45e1f5d9f6ea69/app/controllers/contribute_controller.rb#L40

We extracted the code related to stripping whitespace so that it could be updated if Tufts preferred whitespace handling changes:
https://github.com/curationexperts/epigaea/blob/accdcd100f1cf52fdd221c061a45e1f5d9f6ea69/app/controllers/contribute_controller.rb#L34-L49

I'm sorry if I misunderstood the requirements, but I had written both the ticket title and acceptance criteria to scope the changes to self-deposit items only.  Let us know if you want us to apply the same rules to content that Tisch and DCA staff submits via the administrative edit form.
